<!DOCTYPE html>
<html>
<head>
  <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1">



<title>
  
    Protein Homology Search
  
</title>

<meta name="description" content="小结">
<meta name="keywords" content="Project">
<meta property="og:type" content="article">
<meta property="og:title" content="Protein Homology Search">
<meta property="og:url" content="http://www.yuanenming.com/2018/11/08/Protein-Homology-Search/index.html">
<meta property="og:site_name" content="Yuan Enming">
<meta property="og:description" content="小结">
<meta property="og:locale" content="en">
<meta property="og:updated_time" content="2018-11-11T15:10:38.282Z">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="Protein Homology Search">
<meta name="twitter:description" content="小结">


  <link rel="alternative" href="/atom.xml" title="Yuan Enming" type="application/atom+xml">



  <link rel="icon" href="/images/ming.ico">


<link rel="stylesheet" href="/perfect-scrollbar/css/perfect-scrollbar.min.css">
<link rel="stylesheet" href="/styles/main.css">





</head>
<body
  
    class="monochrome"
  
  >
  <div class="mobile-header">
  <button class="sidebar-toggle" type="button">
    <span class="icon-bar"></span>
    <span class="icon-bar"></span>
    <span class="icon-bar"></span>
  </button>
  <a class="title" href="/">Yuan Enming</a>
</div>

  <div class="main-container">
    <div class="sidebar">
  <div class="header">
    <h1 class="title"><a href="/">Yuan Enming</a></h1>
    
      <p class="subtitle">
        Shanghai Jiao Tong University
      </p>
    
    <div class="info">
      <div class="content">
        
          <div class="description">If you have great talents, industry will improve them; if you have but moderate abilities, industry will supply their deficiency.</div>
        
        
          <div class="author">—Reynolds</div>
        
      </div>
      
        <div class="avatar">
          
            <a href="/about"><img src="/portrait.JPG"></a>
          
        </div>
      
    </div>
  </div>
  <div class="body">
    
      
        <ul class="nav">
          
            
              <li class="tag-list-container">
                <a href="javascript:;">Tag</a>
                <ul class="tag-list"><li class="tag-list-item"><a class="tag-list-link" href="/tags/testing/">-testing</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Algorithm/">Algorithm</a><span class="tag-list-count">13</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Blog/">Blog</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Book/">Book</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Deep-Learning/">Deep Learning</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/LeetCode/">LeetCode</a><span class="tag-list-count">8</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Leetcode/">Leetcode</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Machine-Learning/">Machine Learning</a><span class="tag-list-count">7</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Project/">Project</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/TensorFlow/">TensorFlow</a><span class="tag-list-count">2</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/Tour/">Tour</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/github/">github</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/hexo/">hexo</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/toturials/">toturials</a><span class="tag-list-count">1</span></li><li class="tag-list-item"><a class="tag-list-link" href="/tags/实用/">实用</a><span class="tag-list-count">1</span></li></ul>
              </li>
            
          
            
              <li class="archive-list-container">
                <a href="javascript:;">Archive</a>
                <ul class="archive-list"><li class="archive-list-item"><a class="archive-list-link" href="/archives/2018/">2018</a><span class="archive-list-count">33</span></li></ul>
              </li>
            
          
        </ul>
      
        <ul class="nav">
          
            
              <li>
                <a href="/cv" title="CV">CV</a>
              </li>
            
          
            
              <li>
                <a href="https://github.com/yuanenming" title="Github" target="_blank" rel="noopener">Github</a>
              </li>
            
          
            
              <li>
                <a href="https://weibo.com/u/5022114373" title="weibo" target="_blank" rel="noopener">weibo</a>
              </li>
            
          
            
              <li>
                <a href="https://www.zhihu.com/people/yuan-en-ming/activities" title="zhihu" target="_blank" rel="noopener">zhihu</a>
              </li>
            
          
            
              <li>
                <a href="/yuanenming@sjtu.edu.cn" title="e-mail">e-mail</a>
              </li>
            
          
            
              <li>
                <a href="/mikutap" title="DIY music">DIY music</a>
              </li>
            
          
            
              <li>
                <a href="/game_remove" title="Game">Game</a>
              </li>
            
          
        </ul>
      
    
  </div>
</div>

    <div class="main-content">
      
        <div style="max-width: 1000px">
      
          <article id="post-Protein-Homology-Search" class="article article-type-post">
  
    <h1 class="article-header">
      Protein Homology Search
      
    </h1>
  

  

  <div class="article-info">
    <span class="article-date">
  2018-11-08
</span>

    

    
	<span class="article-tag tagcloud">
		<ul class="article-tag-list"><li class="article-tag-list-item"><a class="article-tag-list-link" href="/tags/Project/">Project</a></li></ul>
	</span>


  </div>
  <div class="article-entry">
  
<div id="toc">
  <ol class="toc"><li class="toc-item toc-level-1"><a class="toc-link" href="#小结-11-10"><span class="toc-text">小结 11.10</span></a><ol class="toc-child"><li class="toc-item toc-level-2"><a class="toc-link" href="#Data"><span class="toc-text">Data</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Baseline-Method"><span class="toc-text">Baseline Method</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Other-Protein-Homology-Search-Software"><span class="toc-text">Other Protein Homology Search Software</span></a></li><li class="toc-item toc-level-2"><a class="toc-link" href="#Future-work"><span class="toc-text">Future work</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#Some-Questions"><span class="toc-text">Some Questions</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Fancy-or-practical"><span class="toc-text">Fancy or practical</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#Schedule"><span class="toc-text">Schedule</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#HMMER3的安装和使用"><span class="toc-text">HMMER3的安装和使用</span></a><ol class="toc-child"><li class="toc-item toc-level-4"><a class="toc-link" href="#HMMER3的安装"><span class="toc-text">HMMER3的安装</span></a></li><li class="toc-item toc-level-4"><a class="toc-link" href="#HMMER3的使用"><span class="toc-text">HMMER3的使用</span></a></li></ol></li></ol></li><li class="toc-item toc-level-2"><a class="toc-link" href="#HHsuite-安装使用"><span class="toc-text">HHsuite 安装使用</span></a></li></ol></li></ol>
</div>

    <p>小结<br><a id="more"></a></p>
<h1 id="小结-11-10"><a href="#小结-11-10" class="headerlink" title="小结 11.10"></a>小结 11.10</h1><p>总结一下最近一周的工作。主要是三个方面。首先是收集和整理数据，从不同的数据源获取数据，并进行整合；第二个是实现基于BLAST和PSI-BLAST的pipeline，这两种方法在很多的研究中作为baseline进行比较，因此可以快速的了解整个问题；第三个是熟悉其他的同源搜索的算法以及相应的软件(<code>HMMER</code>,<code>HHblits</code>)。另外我还开始准备实现PPI网络的baseline。</p>
<h2 id="Data"><a href="#Data" class="headerlink" title="Data"></a>Data</h2><ol>
<li><p><strong>GO Term Information</strong><br>Downloaded all the ontology information from <a href="http://geneontology.org/" target="_blank" rel="noopener">GO</a>, which include <strong>47334</strong> terms in <code>.obo</code> format (released on 2018/10/10).</p>
</li>
<li><p><strong>Protein Sequence Data</strong><br>Downloaded protein sequences data from <a href="ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/" target="_blank" rel="noopener">Swiss-Prot</a>. Uniprot sequence data in <code>fasta</code> format, which include <strong>558590</strong> sequences in total. (released on 2018-10-10)</p>
</li>
<li><p><strong>Annotation Information</strong><br>I spent much time retrieving the GO annotation data. They mainly from the following three sources:</p>
<ul>
<li>Uniprot ID Mapping information<br>This file maps <code>Uniprot ID</code> to various kinds of resources including <code>GO term ID</code>. The original <code>idmapping</code> file contains <strong>134,066,004</strong> (<strong>85,396,292 with GO Annotation</strong>) (Because there are TrEMBL entries). After mapped with Swiss-Prot, There were <strong>527952</strong> remained. Note that the Uniprot annotation was not marked with annotation resources (such as ‘EXP’).</li>
<li><p><a href="ftp://ftp.ebi.ac.uk/pub/databases/GO/goa/UNIPROT/" target="_blank" rel="noopener">GOA</a> data<br>Original annotation file has <strong>181,327,081</strong> entries. After removing entries with less experimental evidence, <strong>822974</strong> remains.</p>
<p>After mapping with Swiss-Prot, <strong>276,853</strong> annotation remains, associated with <strong>44,336</strong> proteins and <strong>22,051</strong> GO terms.</p>
</li>
<li><a href="http://geneontology.org/page/download-go-annotations" target="_blank" rel="noopener">GO</a><br><strong>295,051</strong>annotation entries.</li>
</ul>
</li>
<li><p><strong>PPI information</strong><br>I have downloaded protein-protein interaction networks from <a href="https://string-db.org/cgi/download.pl?sessionId=lZah78NFVD9Y" target="_blank" rel="noopener">STRING</a>. The PPI file contains more than 3,000,000,000 interactions and is large to <strong>123GB</strong>. Now I am tring to map this file to Swiss-Prot.</p>
</li>
</ol>
<hr>
<h2 id="Baseline-Method"><a href="#Baseline-Method" class="headerlink" title="Baseline Method"></a>Baseline Method</h2><p>After retrieving those data, I start to construct a pipline for <strong>BLAST</strong> and <strong>PSI-BLAST</strong> (now trying include <strong>STRING PPI</strong>).<br>The dataflow is as following:</p>
<center>

<div id="flowchart-0" class="flow-chart"></div>
</center>

<p>First the query sequence was searched against the database and obtain multiple sequence alignments (MSA). Then the MSA result can be filtered according to pre-defined homology parameters. And then the filtered MSA is mapped to the GO terms according to a confidence function.</p>
<p>Accoring to the previous study, we can define the confidence score in different ways. For example:<br>if you have a query <code>AKLMNWTY</code> and you get MSA as following:</p>
<figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br></pre></td><td class="code"><pre><span class="line">Query        AKLMNWTY</span><br><span class="line"></span><br><span class="line">Template1    -KICNWT- GO:0000001 evalue=0.01</span><br><span class="line">Template2    --ICNWT- GO:0000001 evalue=0.1</span><br><span class="line">Template3    AKLCNYTY GO:0000002 evalue=0.001</span><br></pre></td></tr></table></figure>
<p>Then you can difine the confidence score of <code>GO:0000001</code> in the following ways:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br></pre></td><td class="code"><pre><span class="line">[1] globalID (identical residues number normalized by target length): </span><br><span class="line">    Cscore(GO:0000001)=4/8</span><br><span class="line">[2] localID (identical residue number normalized by aligned region length):</span><br><span class="line">    Cscore(GO:0000001)=4/6</span><br><span class="line">[3] GOfreq (number of templates annotated with GO divided by template number):</span><br><span class="line">    Cscore(GO:0000001)=2/3</span><br><span class="line">[4] evalue (Sigmoid transform of log evalue):</span><br><span class="line">    Cscore(GO:0000001)=1-Sigmoid(x)=1-1/(1+exp(-x)),x=log(evalue\*alpha)/log(alpha)</span><br></pre></td></tr></table></figure></p>
<p>I have not done much experiment yet, so there is no further analysis.</p>
<hr>
<h2 id="Other-Protein-Homology-Search-Software"><a href="#Other-Protein-Homology-Search-Software" class="headerlink" title="Other Protein Homology Search Software"></a>Other Protein Homology Search Software</h2><p>I have read the papers about <strong>HMMER, HHblits, HMM-HMM comparision</strong>. I think the basic idea of <strong>protein remote homolog search</strong> is make use of as much information about the sequence query and database as possible to achieve a better sensitivity. </p>
<ul>
<li>This is the reason why sequence alignment methods(such as <strong>BLAST</strong>) is inferior to sequence-profile alignment methods(such as <strong>PSI-BLAST</strong>). </li>
<li>The <strong>Sequence Profile</strong> contains each column of a multiple alignment of the protein family, as well as preferred residues at each position which is more conserve than a single character in sequence alignment. </li>
<li><strong>Hidden Markov Models (HMMs)</strong> are similar to simple sequence profiles, but they contain information about the frequency of insertions and deletions at each position, in addition to the amino acid frequencies in the multiple sequence alignment columns. You can use HMMs on the query and the database which both will outperform the sequence-alignment or sequence-profile methods.</li>
<li><strong>HMM-HMM Comparision</strong> exploits HMMs both on the query and the database and is more sensitive to homologs.</li>
</ul>
<p>Then I downloaded a local version of <a href="http://hmmer.org/" target="_blank" rel="noopener">HMMER3.1</a> and <a href="https://github.com/soedinglab/hh-suite/releases" target="_blank" rel="noopener">HHblits</a> and tried to costumize database and parameters.</p>
<h2 id="Future-work"><a href="#Future-work" class="headerlink" title="Future work"></a>Future work</h2><h4 id="Some-Questions"><a href="#Some-Questions" class="headerlink" title="Some Questions"></a>Some Questions</h4><ol>
<li>DeepGO中采用Hierarchical Classification的方法，是利用GO term的层次关系作为restriction，从而试图实现更好的效果。<br>但是我在处理数据的过程中发现，所有数据源的annotation数据，都只是为蛋白序列标注GO层级结构中的叶子节点，也就是并不会使用内部的层次关系。<br>最开始我的一个想法就是探索如何像DeepGO一样利用这个层次结构的关系，但现在开始思考这个方向可行性。<br>另外我通过观察数据得到，一个GO的叶子节点，很有可能与上百个内部节点相关联，这也就意味着，如果考虑内部GO term的话，这个问题将会变得更加复杂.</li>
</ol>
<h4 id="Fancy-or-practical"><a href="#Fancy-or-practical" class="headerlink" title="Fancy or practical"></a>Fancy or practical</h4><ol>
<li>我理解的这个问题是有两个方向，一个是做出一个比较堪用的搜索办法，比如利用<code>HHblits</code>等更敏感的搜索办法，或者像GOLabeler一样引入更多的特征，来实现更好的预测效果。</li>
<li>另一个方向是提出一个fancy的方法，比如我最近看到一篇文章<a href="/https://www.ncbi.nlm.nih.gov/pubmed/29039790">ProLanGO</a>，这篇文章引入了NLP的方法，把sequence到GO的映射理解成机器翻译的过程，protein和GO都看作是一种语言，并以此建模；得到的预测结果虽然一般，但是，整个方法框架还是很出彩的。<br>这两个方向如果能都实现当然最好。</li>
</ol>
<h4 id="Schedule"><a href="#Schedule" class="headerlink" title="Schedule"></a>Schedule</h4><ol>
<li>接下来3-4天继续把baseline的实验结果分析一下，然后，开始尝试PPI与结构信息。</li>
</ol>
<hr>
<h2 id="HMMER3的安装和使用"><a href="#HMMER3的安装和使用" class="headerlink" title="HMMER3的安装和使用"></a>HMMER3的安装和使用</h2><blockquote>
<p>本节参考(<a href="http://blog.sina.com.cn/s/blog_83f77c940101rzlo.html" target="_blank" rel="noopener">http://blog.sina.com.cn/s/blog_83f77c940101rzlo.html</a>)</p>
</blockquote>
<p>简单的讲HMMER就是一款比对软件，可以类似于blast等，但是它提供的比对结果要精确于blast,相应的速度也要慢。与其类似的比对软件有PSI-blast、SAM、PFTOOLS等。</p>
<h4 id="HMMER3的安装"><a href="#HMMER3的安装" class="headerlink" title="HMMER3的安装"></a>HMMER3的安装</h4><p>HMMER3的官网上直接有二进制包，直接下载，根本不需要进行安装，只需要设置一下环境变量就可以使用了</p>
<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br></pre></td><td class="code"><pre><span class="line"><span class="comment"># Linux</span></span><br><span class="line"><span class="comment"># 直接下载匹配OS的二进制包，根本就不需要进行安装，只要稍微设置一下PATH变量就可以使用了，非常方便。</span></span><br><span class="line"><span class="comment"># 在home目录下创建biosoft目录，一般的生物分析软件都在这里</span></span><br><span class="line">mkdir biotools</span><br><span class="line"><span class="built_in">cd</span> biotools</span><br><span class="line"><span class="comment"># 下载二进制包解压，速度好像有些慢，只能无奈等待</span></span><br><span class="line">wget http://eddylab.org/software/hmmer3/3.1b2/hmmer-3.1b2-linux-intel-x86_64.tar.gz</span><br><span class="line">tar -zxvf hmmer-3.1b2-linux-intel-x86_64.tar.gz</span><br><span class="line"><span class="comment"># 将binaries目录添加到环境变量中，能够全局环境使用</span></span><br><span class="line"><span class="built_in">cd</span> ~</span><br><span class="line"><span class="comment"># 添加环境变量</span></span><br><span class="line">vim ./bashrc</span><br><span class="line"><span class="comment">##(添加的语句) export PATH="$PATH:~/biotools/hmmer-3.1b2-linux-intel-x86_64/binaries/"</span></span><br><span class="line"><span class="built_in">source</span> .bashrc</span><br></pre></td></tr></table></figure>
<h4 id="HMMER3的使用"><a href="#HMMER3的使用" class="headerlink" title="HMMER3的使用"></a>HMMER3的使用</h4><p>这个工具包里方法是很全面的，下边主要介绍三种用途。</p>
<ol>
<li>对于一条未知的序列在蛋白质数据库中寻找比对，HMMER完全可以取代目前常用的BLASTP 和 PSI-BLAST。</li>
<li>可以自动注释蛋白质结构域。包括pfam、SMART。</li>
<li>在数据库中寻找已有家族的相似序列。</li>
</ol>
<p>对应与这几个常用的功能（注意每个程序都是输出文件在前面，输入文件在后面）</p>
<ul>
<li><p><strong>hmmbuild</strong> 建立参考数据的隐马尔科夫模型 profile HMM construction from multiple sequence alignments</p>
<ul>
<li>输入：多重序列比对文件（Stockholm file）</li>
<li>输出：建立这些多重序列比对的隐马尔科夫模型</li>
<li>example:<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br><span class="line">2</span><br><span class="line">3</span><br><span class="line">4</span><br><span class="line">5</span><br><span class="line">6</span><br><span class="line">7</span><br><span class="line">8</span><br><span class="line">9</span><br><span class="line">10</span><br><span class="line">11</span><br><span class="line">12</span><br><span class="line">13</span><br><span class="line">14</span><br><span class="line">15</span><br><span class="line">16</span><br></pre></td><td class="code"><pre><span class="line">$ hmmbuild globins4.hmm tutorial/globins4.sto</span><br><span class="line"></span><br><span class="line"><span class="comment"># hmmbuild :: profile HMM construction fmultiple    sequence alignments</span></span><br><span class="line"><span class="comment"># HMMER 3.1b2 (February 2015); http://hmmer.org/</span></span><br><span class="line"><span class="comment"># Copyright (C) 2015 Howard Hughes Medical Institute.</span></span><br><span class="line"><span class="comment"># Freely distributed under the GNU General PubLicense ( GPLv3).</span></span><br><span class="line"><span class="comment"># - - - - - - - - - - - - - - - - - - - - - - - - - -   - - - - - - - -</span></span><br><span class="line"><span class="comment"># input alignment file:             tutorglobins4.sto</span></span><br><span class="line"><span class="comment"># output HMM file:                  globins4.hmm</span></span><br><span class="line"><span class="comment"># - - - - - - - - - - - - - - - - - - - - - - - - - -   - - - - - - - -</span></span><br><span class="line"></span><br><span class="line"><span class="comment"># idx name                  nseq  alen  mlen eff_nre/   pos description</span></span><br><span class="line"><span class="comment">#---- -------------------- ----- ----- ----------   ------ -----------</span></span><br><span class="line">1     globins4                 4   171   149 0.96   0.589</span><br><span class="line"></span><br><span class="line"><span class="comment"># CPU time: 0.15u 0.00s 00:00:00.15 Elaps00:00:00.31</span></span><br></pre></td></tr></table></figure>
</li>
</ul>
</li>
<li><p><strong>hmmalign</strong> 多重序列比对 align sequences to a profile HMM</p>
<ul>
<li>输入：输入是fasta文件和建立的模型文件</li>
<li>输出：Stockholm 格式的多重序列比对文件</li>
<li>example:<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">$ hmmalign globins4.hmm tutorial/globins45.fa</span><br></pre></td></tr></table></figure>
</li>
</ul>
</li>
<li><p><strong>hmmsearch</strong> search profile(s) against a sequence database</p>
<ul>
<li>输入：建立好的参考数据模型、要搜索的数据库</li>
<li>输出：输出结果</li>
<li>example:<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hmmsearch globins4.hmm uniprot sprot.fasta &gt; globins4.out</span><br></pre></td></tr></table></figure>
</li>
</ul>
</li>
<li><p><strong>phmmer</strong> search a protein sequence against a protein database<br>  这是blast的功能。对于只有一条未知蛋白序列的情况下，就省去多重序列比对和建立模型的过程直接一步到位，期间使用了BLOSUM62 scores打分矩阵，类似于BLASTP-like）</p>
<ul>
<li>输入:tutorial/HBB HUMAN 是你要检测的fasta格式的序列；uniprot sprot.fasta你要搜索的数据库<figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">phmmer tutorial/HBB HUMAN uniprot sprot.fasta</span><br></pre></td></tr></table></figure>
</li>
</ul>
</li>
<li><p><strong>nhmmer</strong> 类似的就有</p>
</li>
</ul>
<hr>
<p>如果你有未知的序列去搜寻已知的模型数据库，例如：Pfam, SMART, or TIGRFams.这个时候搜寻的过程就使用hmmscan,相应的核酸序列就使用nhmmscan.<br>首先还是要建立模型数据库，如果你要合并多个可以直接:<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">cat globins4.hmm fn3.hmm Pkinase.hmm &gt; minifam</span><br></pre></td></tr></table></figure></p>
<p>如果你仅仅有的是模型数据库的多重序列比对格式文件例如来自pfam数据库的（Stockholm file）Pfam-A.seed文件：你可以利用hmmbuild 建立模型数据库：<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hmmbuild Pfam-A.hmm Pfam-A.see</span><br></pre></td></tr></table></figure></p>
<p>生成的文件为pfam-A.hmm<br>由于你一般建立的模型数据库比较大，所以需要建立index,运行：<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hmmpress minifam <span class="comment">#（感觉有点像formatdb）</span></span><br></pre></td></tr></table></figure></p>
<p>最后运行：<br><figure class="highlight bash"><table><tr><td class="gutter"><pre><span class="line">1</span><br></pre></td><td class="code"><pre><span class="line">hmmscan minifam tutorial/7LESS DROME</span><br></pre></td></tr></table></figure></p>
<h2 id="HHsuite-安装使用"><a href="#HHsuite-安装使用" class="headerlink" title="HHsuite 安装使用"></a>HHsuite 安装使用</h2><p><script src="https://cdnjs.cloudflare.com/ajax/libs/raphael/2.2.7/raphael.min.js"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/flowchart/1.6.5/flowchart.min.js"></script><textarea id="flowchart-0-code" style="display: none">st=>start: Query
op1=>operation: MSA
op2=>operation: filtered_MSA
e=>end: GO Term

st->op1->op2->e</textarea><textarea id="flowchart-0-options" style="display: none">{"scale":1,"line-width":2,"line-length":50,"text-margin":10,"font-size":12}</textarea><script>  var code = document.getElementById("flowchart-0-code").value;  var options = JSON.parse(decodeURIComponent(document.getElementById("flowchart-0-options").value));  var diagram = flowchart.parse(code);  diagram.drawSVG("flowchart-0", options);</script></p>

  </div>
  <footer class="article-footer">
    
  <div class="cc">
    <a href="http://creativecommons.org/licenses/by-sa/4.0/deed.e" target="_blank" title="Attribution-ShareAlike">
      <img src="/images/cc/cc.png">
      
          <img src="/images/cc/by.png">
        
          <img src="/images/cc/sa.png">
      
      <span>
        This work is licensed under a Creative Commons Attribution-ShareAlike 4.0 International License.
      </span>
    </a>
  </div>


    

  </footer>
  <footer class="article-footer">
  
      <section class="livere" id="comments">
    <!-- 来必力City版安装代码 -->
<div id="lv-container" data-id="city" data-uid="MTAyMC8zODI5Ny8xNDgyNQ==">
 <script type="text/javascript">
   (function(d, s) {
       var j, e = d.getElementsByTagName(s)[0];

       if (typeof LivereTower === 'function') { return; }

       j = d.createElement(s);
       j.src = 'https://cdn-city.livere.com/js/embed.dist.js';
       j.async = true;

       e.parentNode.insertBefore(j, e);
   })(document, 'script');
 </script>
<noscript> 为正常使用来必力评论功能请激活JavaScript</noscript>
</div>
<!-- City版安装代码已完成 -->
</section>
    
  </footer>
</article>







          <div align="center" class="main-footer">
<!-- 根据页面mathjax变量决定是否加载MathJax数学公式js -->

<span id="timeDate">载入天数...</span><span id="times">载入时分秒...</span>
<script>
    var now = new Date(); 
    function createtime() { 
        var grt= new Date("07/20/2018 12:49:00");//此处修改你的建站时间或者网站上线时间 
        now.setTime(now.getTime()+250); 
        days = (now - grt ) / 1000 / 60 / 60 / 24; dnum = Math.floor(days); 
        hours = (now - grt ) / 1000 / 60 / 60 - (24 * dnum); hnum = Math.floor(hours); 
        if(String(hnum).length ==1 ){hnum = "0" + hnum;} minutes = (now - grt ) / 1000 /60 - (24 * 60 * dnum) - (60 * hnum); 
        mnum = Math.floor(minutes); if(String(mnum).length ==1 ){mnum = "0" + mnum;} 
        seconds = (now - grt ) / 1000 - (24 * 60 * 60 * dnum) - (60 * 60 * hnum) - (60 * mnum); 
        snum = Math.round(seconds); if(String(snum).length ==1 ){snum = "0" + snum;} 
        document.getElementById("timeDate").innerHTML = "本站已安全运行 "+dnum+" 天 "; 
        document.getElementById("times").innerHTML = hnum + " 小时 " + mnum + " 分 " + snum + " 秒"; 
    } 
setInterval("createtime()",250);
</script>
  
    © 2018 Yuan Enming - Powered by <a href="http://hexo.io" target="_blank">Hexo</a> - Theme <a href="https://github.com/denjones/hexo-theme-chan" target="_blank">Chan</a>
  
  <script type="text/javascript">
  (function(w,d,t,u,n,s,e){w['SwiftypeObject']=n;w[n]=w[n]||function(){
  (w[n].q=w[n].q||[]).push(arguments);};s=d.createElement(t);
  e=d.getElementsByTagName(t)[0];s.async=1;s.src=u;e.parentNode.insertBefore(s,e);
  })(window,document,'script','//s.swiftypecdn.com/install/v2/st.js','_st');
  
  _st('install','iCyu54VhBnr8QxsJnwf6','2.0.0');
</script>
</div>

      
        </div>
      
    </div>
  </div>
  <script src="//apps.bdimg.com/libs/jquery/2.1.4/jquery.min.js"></script>

  <link rel="stylesheet" href="/PhotoSwipe/photoswipe.css">
  <link rel="stylesheet" href="/PhotoSwipe/default-skin/default-skin.css">

  <!-- Root element of PhotoSwipe. Must have class pswp. -->
  <div class="pswp" tabindex="-1" role="dialog" aria-hidden="true">

    <!-- Background of PhotoSwipe.
             It's a separate element as animating opacity is faster than rgba(). -->
    <div class="pswp__bg"></div>

    <!-- Slides wrapper with overflow:hidden. -->
    <div class="pswp__scroll-wrap">

      <!-- Container that holds slides.
                PhotoSwipe keeps only 3 of them in the DOM to save memory.
                Don't modify these 3 pswp__item elements, data is added later on. -->
      <div class="pswp__container">
        <div class="pswp__item"></div>
        <div class="pswp__item"></div>
        <div class="pswp__item"></div>
      </div>

      <!-- Default (PhotoSwipeUI_Default) interface on top of sliding area. Can be changed. -->
      <div class="pswp__ui pswp__ui--hidden">

        <div class="pswp__top-bar">

          <!--  Controls are self-explanatory. Order can be changed. -->

          <div class="pswp__counter"></div>

          <button class="pswp__button pswp__button--close" title="Close (Esc)"></button>

          <button class="pswp__button pswp__button--share" title="Share"></button>

          <button class="pswp__button pswp__button--fs" title="Toggle fullscreen"></button>

          <button class="pswp__button pswp__button--zoom" title="Zoom in/out"></button>

          <!-- Preloader demo http://codepen.io/dimsemenov/pen/yyBWoR -->
          <!-- element will get class pswp__preloader--active when preloader is running -->
          <div class="pswp__preloader">
            <div class="pswp__preloader__icn">
              <div class="pswp__preloader__cut">
                <div class="pswp__preloader__donut"></div>
              </div>
            </div>
          </div>
        </div>

        <div class="pswp__share-modal pswp__share-modal--hidden pswp__single-tap">
          <div class="pswp__share-tooltip"></div>
        </div>

        <button class="pswp__button pswp__button--arrow--left" title="Previous (arrow left)"></button>

        <button class="pswp__button pswp__button--arrow--right" title="Next (arrow right)"></button>

        <div class="pswp__caption">
          <div class="pswp__caption__center"></div>
        </div>
      </div>
    </div>
  </div>

  <script src="/PhotoSwipe/photoswipe.js"></script>
  <script src="/PhotoSwipe/photoswipe-ui-default.js"></script>


<script src="/perfect-scrollbar/js/min/perfect-scrollbar.min.js"></script>
<script src="/scripts/main.js"></script>

<script src="/live2dw/lib/L2Dwidget.min.js?0c58a1486de42ac6cc1c59c7d98ae887"></script><script>L2Dwidget.init({"log":false,"pluginJsPath":"lib/","pluginModelPath":"assets/","pluginRootPath":"live2dw/","tagMode":false});</script></body>
</html>
